src/borg/cache.py

```diff
@@ -900,6 +909,13 @@ def create_master_idx(chunk_idx):
             self.do_cache = os.path.isdir(archive_path)
             self.chunks = create_master_idx(self.chunks)
 
+    def check_cache_integrity(self):
```

Comment:
this only tests existence, not integrity.

how about:
```
return all(os.path.exists(os.path.join(self.path, file)) for file in "chunks", "config")
```

Reply:
also: only called from one place.

maybe rather do the joins only once and have the rest in-line?